kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -607,6 +658,54 @@ public <M extends ApiMessage> CompletionStage<M> sendRequest(RequestHeaderData h
             return filterPromise.minimalCompletionStage();
         }
 
+        @Override
+        public CompletionStage<Map<Uuid, String>> getTopicNames(Set<Uuid> topicIds) {
+            Objects.requireNonNull(topicIds);
+            return requestTopicMetadata(topicIds).thenApply(f -> extractTopicNames(topicIds, f));
+        }
+
+        private CompletionStage<ApiMessage> requestTopicMetadata(Set<Uuid> topicIds) {
+            MetadataRequestData request = new MetadataRequestData();
+            request.setTopics(topicIds.stream().map(topicId -> new MetadataRequestData.MetadataRequestTopic().setTopicId(topicId)).toList());
+            request.setAllowAutoTopicCreation(false);
+            request.setIncludeClusterAuthorizedOperations(false);
+            request.setIncludeTopicAuthorizedOperations(false);
+            RequestHeaderData requestHeaderData = new RequestHeaderData();
+            requestHeaderData.setRequestApiKey(ApiKeys.METADATA.id);
+            requestHeaderData.setRequestApiVersion(METADATA_API_VER_WITH_TOPIC_ID_SUPPORT);
+            return sendRequest(requestHeaderData, request, true);
```

Comment:
the `true` is where we are saying the response is cacheable, this is the only internal request marked like this.